### DISCUSSION TITLE
Getting a character in the foreground when using two ControlNets

### DISCUSSION THREAD

**Discussion by simonknittel** [lllyasviel/ControlNet#501] [Q&A]
Hey!

I'm trying to generate a character sitting on top a planet. So far I've got the pose and planet nailed down thanks to the OpenPose and LineArt models. Also the general composition and size of elements is correct. However, the character gets rendered behind the planet. I would like to have it in front/on top of the character.

Is there a trick to achieve that?

**Example**
![sana](https://github.com/lllyasviel/ControlNet/assets/8451099/37035099-44e9-40a1-ba63-85e125536a63)

**ControlNet Nodes in the ComfyUI Workflow**
![image](https://github.com/lllyasviel/ControlNet/assets/8451099/e8110db3-79c5-4ce8-bb83-73ab7499caf5)

**Completel Workflow**
[workflow.zip](https://github.com/lllyasviel/ControlNet/files/12321126/workflow.zip)

I've already tried to be more specific in the prompt. I tried to play around with the strenghts of the ControlNet nodes. Nothing really changed.

**Comment by dayunbao** [comment]
I asked a similar question in the sd-webui-controlnet repo [here](https://github.com/Mikubill/sd-webui-controlnet/discussions/1853).  The maintainer of that repo built a new external tool that will allow you to do that.

**Comment by geroldmeisinger** [comment]
why do want to achieve everything with Stable Diffusion alone? couldn't you just generate the images separatly and merge them in Gimp or something.
